XRandR: Allow multiple but not all CRTCs to be redshifted

lennartS opened the pull request:
Previously only one CRTC could be set in the configuration file for
redshifting when XRandR mechanism was being used. That is fine for
a setup with two displays but breaks when three or more displays
are in use and one of those shouldn't be redshifted (e.g. two
computer displays and one TV connected to the computer).

The config value 'crtc' for method xrandr can now be entered as
comma separated list of multiple CRTCs. All CRTCs in the list will
be redshifted while all those not in the list will not be touched.

Signed-off-by: Lennart Sauerbeck devel@lennart.sauerbeck.org

davorb commented:
@lennartS has this functionality been added to redshift?

lennartS commented:
@davorb Sorry, but I don't think so. I'll re-open this for now. I was just doing some housekeeping and saw this was still open.

davorb commented:
Thanks. I would love for this functionality to be added, but it seems like redshift might be abandoned. There have been no commits since February.

jonls commented:
@davorb Definitely not abandoned. Though, the time that I have been able to invest lately has been very limited. Let me know if you are interested in getting the code back on track for merge.

lennartS commented:
@jonls Do you mean the patch would be acceptable in principle now? If so, I can take a look at it to get it mergable again.

jonls commented:
@lennartS Yes, the other solution that was proposed did not move forward. I would like to merge this instead.

lennartS commented:
@jonls That's great to hear. I rebased my change onto your master and fixed the conflicts. Quick testing didn't reveal any bugs on my machine, it still works the same as before the rebase.

jonls commented:
@lennartS I made some changes to your pull request to make parsing the list of CRTCs more robust and to provide an error message when the CRTC numbers are incorrectly specified. Can you please test the code in #421?

lennartS commented:
@jonls I confirmed that #421 works as expected and commented over there as well. You added some good changes to my original patch :)

Should I close this pull request now?

jonls commented:
@lennartS 👍  Merged.
